src/orion/client/experiment.py

```diff
@@ -738,12 +732,20 @@ def workon(
             If the callblack returns False, the error will be ignored, otherwise it is counted
             for the threshold `max_broken`. In case of critical errors, you may also directly
             raise an error and force break out of ``workon``.
+        idle_timeout: int, optional
+            Maximum time (seconds) allowed for idle workers. LazyWorkers will be raised if
+            timeout is reached. Such timeout are generally caused when reaching the
+            end of the optimization when no new trials can be sampled for the idle workers.
+            Defaults to ``orion.core.config.worker.idle_timeout``.
         **kwargs
             Constant argument to pass to `fct` in addition to trial.params. If values in kwargs are
             present in trial.params, the latter takes precedence.
 
         Raises
         ------
+        :class:`orion.core.utils.exceptions.LazyWorkers`
+             If workers stay idle for too long.
+
         :class:`orion.core.utils.exceptions.InvalidResult`
              If results returned by `fct` have invalid format.
 
```

Comment:
Can it still raise ReservationTimeout? I think it is ReservationRaceCondition now, but even that should be caught by the  runner and not be raised up to workon().

Reply:
fixed